build.gradle

```diff
@@ -61,23 +62,23 @@ sourceSets {
         compileClasspath += sourceSets.api.output
         runtimeClasspath += sourceSets.api.output
     }
-    additions { setupSourceSet(sourceSets.additions, 'additions') }
-    generators { setupSourceSet(sourceSets.generators, 'generators') }
-    defense { setupSourceSet(sourceSets.defense, 'defense') }
-    tools { setupSourceSet(sourceSets.tools, 'tools') }
+    additions { setupSourceSet((SourceSet) sourceSets.additions, 'additions') }
+    generators { setupSourceSet((SourceSet) sourceSets.generators, 'generators') }
+    defense { setupSourceSet((SourceSet) sourceSets.defense, 'defense') }
+    tools { setupSourceSet((SourceSet) sourceSets.tools, 'tools') }
     test {
         compileClasspath += sourceSets.api.output + sourceSets.main.output + sourceSets.additions.output + sourceSets.generators.output + sourceSets.defense.output + sourceSets.tools.output
         runtimeClasspath += sourceSets.api.output + sourceSets.main.output + sourceSets.additions.output + sourceSets.generators.output + sourceSets.defense.output + sourceSets.tools.output
     }
     //Data gen modules for the different modules
-    datagenmain { setupDataGenSourceSet(sourceSets.datagenmain, null, 'main') }
-    datagenadditions { setupDataGenSourceSet(sourceSets.datagenadditions, sourceSets.additions, 'additions') }
-    datagengenerators { setupDataGenSourceSet(sourceSets.datagengenerators, sourceSets.generators, 'generators') }
-    datagendefense { setupDataGenSourceSet(sourceSets.datagendefense, sourceSets.defense, 'defense') }
-    datagentools { setupDataGenSourceSet(sourceSets.datagentools, sourceSets.tools, 'tools') }
+    datagenmain { setupDataGenSourceSet((SourceSet) sourceSets.datagenmain, null, 'main') }
+    datagenadditions { setupDataGenSourceSet((SourceSet) sourceSets.datagenadditions, (SourceSet) sourceSets.additions, 'additions') }
+    datagengenerators { setupDataGenSourceSet((SourceSet) sourceSets.datagengenerators, (SourceSet) sourceSets.generators, 'generators') }
+    datagendefense { setupDataGenSourceSet((SourceSet) sourceSets.datagendefense, (SourceSet) sourceSets.defense, 'defense') }
+    datagentools { setupDataGenSourceSet((SourceSet) sourceSets.datagentools, (SourceSet) sourceSets.tools, 'tools') }
 }
 
-//This method sets up an additional sourceset in src/$name and adds a reference to the corresponding
+//This method sets up an additional sourceSet in src/$name and adds a reference to the corresponding
```

Comment:
In this section, the casts seem to be able to be replaced by a short ref (i.e. instead of `sourceSets.foo`, just `foo`). Seems to import to IDEA just fine